File: fix_services_final.js
import fs from "fs";

// Read the file
let content = fs.readFileSync("/workspace/app/data/services.ts", "utf8");

// Count how many patterns we need to fix
const pattern1 = /users: ([^,]+),}freeTrial: ([^,]+),},/g;
const pattern2 = /users: ([^,]+),}freeTrial: ([^,]+),}/g;

let matches1 = [...content.matchAll(pattern1)];
let matches2 = [...content.matchAll(pattern2)];

console.log(`Found ${matches1.length} matches for pattern 1`);
console.log(`Found ${matches2.length} matches for pattern 2`);

// Fix pattern 1: users: 'X',}freeTrial: 'Y',},
content = content.replace(
  /users: ([^,]+),}freeTrial: ([^,]+),},/g,
  (match, users, freeTrial) => {
    return `users: ${users},\n    freeTrial: ${freeTrial}\n  },`;
  },
);

// Fix pattern 2: users: 'X',}freeTrial: 'Y',}
content = content.replace(
  /users: ([^,]+),}freeTrial: ([^,]+),}/g,
  (match, users, freeTrial) => {
    return `users: ${users},\n    freeTrial: ${freeTrial}\n  }`;
  },
);

// Write the fixed content back
fs.writeFileSync("/workspace/app/data/services.ts", content);

console.log("Fixed services.ts file");

File: fix_services_final.py
#!/usr/bin/env python3

import re

# Read the file
with open('/workspace/app/data/services.ts', 'r') as f:
    content = f.read()

# Count patterns before fixing
pattern1_count = len(re.findall(r"users: [^,]+,\}freeTrial: [^,]+,\},", content))
pattern2_count = len(re.findall(r"users: [^,]+,\}freeTrial: [^,]+,\}", content))

print(f"Found {pattern1_count} matches for pattern 1")
print(f"Found {pattern2_count} matches for pattern 2")

# Fix the malformed object syntax
# Pattern: users: 'X',}freeTrial: 'Y',},
content = re.sub(r"users: ([^,]+),\}freeTrial: ([^,]+),\},", r"users: \1,\n    freeTrial: \2\n  },", content)

# Pattern: users: 'X',}freeTrial: 'Y',}
content = re.sub(r"users: ([^,]+),\}freeTrial: ([^,]+),\}", r"users: \1,\n    freeTrial: \2\n  }", content)

# Write the fixed content back
with open('/workspace/app/data/services.ts', 'w') as f:
    f.write(content)

print("Fixed services.ts file with correct patterns")